homeassistant/components/atag/__init__.py

```diff
@@ -75,27 +74,12 @@ def __init__(self, coordinator: DataUpdateCoordinator, atag_id: str) -> None:
         super().__init__(coordinator)
 
         self._id = atag_id
-        self._name = DOMAIN.title()
-
-    @property
-    def device_info(self) -> DeviceInfo:
-        """Return info for device registry."""
-        device = self.coordinator.data.id
-        version = self.coordinator.data.apiversion
-        return {
-            "identifiers": {(DOMAIN, device)},
+        self._attr_name = DOMAIN.title()
+        self._attr_unique_id = f"{coordinator.data.id}-{atag_id}"
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, coordinator.data.id)},
             "name": "Atag Thermostat",
             "model": "Atag One",
-            "sw_version": version,
+            "sw_version": coordinator.data.apiversion,
```

Comment:
This changes the behavior of the integration. Previously, this would update when the firmware of the device updated, now it will only update when the integration is set up.